visualization/src/point_cloud_handlers.cpp

```diff
@@ -686,56 +686,53 @@ pcl::visualization::PointCloudGeometryHandler<pcl::PCLPointCloud2>::getGeometry
 
   vtkSmartPointer<vtkFloatArray> data = vtkSmartPointer<vtkFloatArray>::New ();
   data->SetNumberOfComponents (3);
+  
   vtkIdType nr_points = cloud_->width * cloud_->height;
+  
+  if (!data->Resize(nr_points))
+    PCL_ERROR("Failed to allocate space for points in VTK array");
 
   // Add all points
-  float dim;
-  vtkIdType j = 0;    // true point index
-  std::vector<float> pts (nr_points * 3);
+  float x,y,z;
   int point_offset = 0;
 
   // If the dataset has no invalid values, just copy all of them
   if (cloud_->is_dense)
   {
     for (vtkIdType i = 0; i < nr_points; ++i, point_offset+=cloud_->point_step)
-    {
+    {      
       // Copy the value at the specified field
-      memcpy (&dim, &cloud_->data[point_offset + cloud_->fields[field_x_idx_].offset], sizeof (float));
-      pts[i * 3 + 0] = dim;
+      memcpy (&x, &cloud_->data[point_offset + cloud_->fields[field_x_idx_].offset], sizeof (float));
```

Comment:
Any chance we can reduce this to a simple float assignment instead of calling memcpy?

Reply:
I did think about it, but I'm unsure how to correctly cast the _data (uint_8) arraypointer

Reply:
Something like this - not sure if thats prettier.
```
const float* xPointer = (const float*)&cloud_->data[point_offset + cloud_->fields[field_x_idx_].offset];
data->InsertNextValue(xPointer[0]);
```

Reply:
Yes, but use `static_cast`

Reply:
I'm getting invalid type conversion, when using static_cast?

Reply:
reinterpret_cast seems to work?

Reply:
Try static first (likely won't work) and move to reinterpret if the first one fails
```cpp
const float* ptr = static_cast<const float*>(&cloud_->data[point_offset + cloud_->fields[field_x_idx_].offset]);
data->InsertNextValue(*ptr);
```